src/game/server/baseentity.cpp

```diff
@@ -1698,9 +1698,9 @@ int CBaseEntity::TakeDamage( const CTakeDamageInfo &inputInfo )
 		{
 			info.ScaleDamage( info.GetAttacker()->GetAttackDamageScale( this ) );
 		}
-
+		
 		// Scale the damage by my own modifiers
-		info.ScaleDamage( GetReceivedDamageScale( info.GetAttacker() ) );
+		info.ScaleDamage( GetReceivedDamageScale( info.GetAttacker() ? info.GetAttacker() : NULL ) );
```

Comment:
if you want to be extra safe, `info && info.GetAttacker() ? info.GetAttacker() : NULL`